atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueServiceTest.java

```diff
@@ -972,6 +972,63 @@ public void shouldAllowRemovingAllCellsInDynamicColumns() {
         assertThat(values, is(emptyMap()));
     }
 
+
+    @Test
+    public void shouldAllowSameTablenameDifferentNamespace() {
+        TableReference fooBar = TableReference.createUnsafe("foo.bar");
+        TableReference bazBar = TableReference.createUnsafe("baz.bar");
+
+        // try create table in same call
+        keyValueService.createTables(
+                ImmutableMap.of(
+                        fooBar, AtlasDbConstants.GENERIC_TABLE_METADATA,
+                        bazBar, AtlasDbConstants.GENERIC_TABLE_METADATA));
+
+        // try create table spanned over different calls
+        keyValueService.createTable(fooBar, AtlasDbConstants.GENERIC_TABLE_METADATA);
+        keyValueService.createTable(bazBar, AtlasDbConstants.GENERIC_TABLE_METADATA);
+
+        // test table actually created
+        keyValueService.getRange(fooBar, RangeRequest.all(), 0L);
+        keyValueService.getRange(bazBar, RangeRequest.all(), 0L);
+
+        // clean up
+        keyValueService.dropTables(ImmutableSet.of(fooBar, bazBar));
+    }
+
+    @Test
+    public void truncateShouldBeIdempotent() {
+        TableReference fooBar = TableReference.createUnsafe("foo.bar");
+        keyValueService.createTable(fooBar, AtlasDbConstants.GENERIC_TABLE_METADATA);
+
+        keyValueService.truncateTable(fooBar);
+        keyValueService.truncateTable(fooBar);
+
+        keyValueService.dropTable(fooBar);
+    }
+
+    @Test
+    public void truncateOfNonExistantTableShouldThrow() {
+        try {
+            keyValueService.truncateTable(TEST_NONEXISTING_TABLE);
+            Assert.fail("truncate must throw on failure");
+        } catch (RuntimeException e) {
+            // expected
+        }
+    }
+
+    @Test
+    public void dropTableShouldBeIdempotent() {
```

Comment:
the table that we're dropping never existed in the first place, which puts us mostly in the same state as calling drop twice.

Because the state isn't completely identical, I'm going to agree with you.